Assets/VRTK/Scripts/VRTK_PlayerJump.cs

```diff
@@ -0,0 +1,366 @@
+    /// 
+    /// Jump scaling for non-fixed jump factors is calculated in CalcAdjustedFinalJumpVelocity
+    /// </remarks>
+    /// <example>
+    /// VRTK/Examples/039_CameraRig_PlayerJump
+    /// </example>
+    public class VRTK_PlayerJump : MonoBehaviour
+    {
+        #region ### Public Properties ###
+
+        [Tooltip("Jump velocity in the up direction is magnified by this")]
+        public float upFactor = 5.0f;
+
+        [Tooltip("Jump velocity in the facing direction is magnified by this")]
+        public float forwardFactor = 3.0f;
+
+        [Tooltip("Offset to detect a crouch before jump")]
+        public float downThreshold = -0.8f;
+
+        [Tooltip("Offset to detect a spring into jump")]
+        public float upThreshold = 1.0f;
+
+        [Tooltip("Time in seconds to reset")]
+        public float timeout = 1.0f;
+
+        [Tooltip("Fixed vs dynamic jump power - uncheck to scale based on player jump intensity")]
+        public bool jumpFixed = false;
+
+        [Tooltip("Amount to magnify the player jump intensity when jump scaling not fixed - modifies Up Factor and Forward Factor")]
+        public float jumpScaling = 3.0f;
+
+        #endregion
+
+        #region ### Private Properties ###
+
+        // deviations less than this will not be counted against baseline establishment - magic number alert!
+        private float baselineUpdateSensitivity = 0.5f;
+
+        // this is the baseline height recorded from the HMD over time
+        private float baselineHeight = 0;
+
+        // this is a time since starting crouch.  Zero if not in jump state
+        private float crouchStartTime = 0;
+
+        // player jumping state as understood by this script
+        private bool jumping = false;
+
+        // previous height of HMD;
+        private float lastHeight;
+
+        // lowest point in crouch
+        private float crouchLow;
+
+        // time of crouchLow
+        private float crouchLowTime;
+
+        // velocity history
+        private float[] historicVelocity;
+
+        // number of history points to compare
+        private int historyLen = 5;
+
+        private Transform HMDTransform;
+        private VRTK_PlayerPresence playerPresence;
+
+        #endregion
+
+        #region ### private logging vars ###
+
+        // set to writable directory eg. C:/Users/Public/Documents/Unity Projects/MyProject/Assets
+        private string loggingTarget = "";
+        private StreamWriter fileWriter;
+        private bool logging = false;
+        private bool debugging = true;
+
+        #endregion
+
+        private void Awake()
+        {
+            // Required Component: VRTK_PlayerPresence
+            playerPresence = GetComponent<VRTK_PlayerPresence>();
+            if (!playerPresence)
+            {
+                playerPresence = gameObject.AddComponent<VRTK_PlayerPresence>();
```

Comment:
If a PlayerPresence is always required then use the [RequiredComponent] unity annotation instead so less magic happens at runtime